src/cartesian/YAxis.tsx

```diff
@@ -111,20 +128,23 @@ const YAxisImpl: FunctionComponent<Props> = (props: Props) => {
     width,
   ]);
 
-  if (axisSize == null || position == null) {
+  if (axisSize == null || position == null || synchronizedSettings == null) {
     return null;
   }
 
   const { dangerouslySetInnerHTML, ticks, ...allOtherProps } = props;
+  const { id, ...restSynchronizedSettings } = synchronizedSettings;
 
   return (
     <CartesianAxis
       {...allOtherProps}
+      {...restSynchronizedSettings}
       ref={cartesianAxisRef}
       labelRef={labelRef}
       scale={scale}
       x={position.x}
       y={position.y}
+      tickTextProps={width === 'auto' ? { width: undefined } : { width }}
```

Comment:
If width is auto then we need to let Text calculate its own width, so we overwrite whatever width is coming from the axisProps